### ISSUE TITLE
Different result between string solver z3str3 and seq

### ISSUE THREAD

**Issue by drogonFan** [Z3Prover/z3#5278]
We found a SMT instance , the default string solver of Z3(seq) and cvc4 both returns `SAT`,  but Z3STR3 returns `UNSAT`

test.smt2:
```
(set-logic QF_S)
(declare-fun var3 () Int)
(declare-fun var4 () Int)
(declare-fun var0 () String)
(declare-fun var1 () String)
(assert (str.prefixof var1 var1))
(assert (>= var4 var3))
(assert (>= var4 25))
(assert (str.suffixof (str.replace (str.replace var1 var1 var0) (str.replace var0 "Lgw!'\t'P?Db?" var0) (str.at var0 40)) (str.replace (str.at "eh5-+iv1b25m" var4) (str.at "y8H0C^zM" 14) (str.substr var1 0 40))))
(assert (str.prefixof var0 "@d:obAw-x;"))
(assert (str.prefixof var1 "qWs'1(>i}g"))
(check-sat)

```

System version：
```
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.2 LTS
Release:	20.04
```
z3 version：
```
Z3 version 4.8.11 - 64 bit
```

The execution result：
```
z3 smt.string_solver=seq test.smt2
sat

z3 smt.string_solver=z3str3 test.smt2
unsat

cvc4 --strings-exp test.smt2
sat
```

**Comment by drogonFan** [comment]
sat model :

```
sat
(model
(define-fun var3 () Int 0)
(define-fun var4 () Int 25)
(define-fun var0 () String "")
(define-fun var1 () String "")
)
```

When I put the model into the test case,

```
(set-logic QF_S)
(declare-fun var3 () Int)
(declare-fun var4 () Int)
(declare-fun var0 () String)
(declare-fun var1 () String)
(assert (str.prefixof "" ""))
(assert (>= 25 0))
(assert (>= 25 25))
(assert (str.suffixof (str.replace (str.replace "" "" "") (str.replace "" "Lgw!'\t'P?Db?" "") (str.at "" 40)) (str.replace (str.at "eh5-+iv1b25m" 25) (str.at "y8H0C^zM" 14) (str.substr "" 0 40))))
(assert (str.prefixof "" "@d:obAw-x;"))
(assert (str.prefixof "" "qWs'1(>i}g"))
(check-sat)
```

z3str3 return `SAT`

```
z3 smt.string_solver=z3str3 test2.smt2
sat
```

**Comment by JohnLyu2** [comment]
The z3str3 bug is fixed in #6000.
